nautobot/core/tests/runner.py

```diff
@@ -30,3 +40,87 @@ def __init__(self, **kwargs):
             kwargs["exclude_tags"] = incoming_exclude_tags
 
         super().__init__(**kwargs)
+
+    def generate_report(self, test_results, result):
+        """
+        Generate Performance Report consists of unittests that are significantly slower than baseline.
+        """
+        test_result_count = len(test_results)
+
+        # Add `--generate_report` to the end of `invoke` commands to generate a report.json file consist of the performance tests result
+        if self.report_path:
+            data = {
+                "tests": [{"name": func_name, "execution_time": float(timing)} for func_name, timing in test_results],
+                "nb_tests": result.testsRun,
+                "nb_failed": len(result.errors + result.failures),
+                "total_execution_time": result.timeTaken,
+            }
+            with open(self.report_path, "w") as outfile:
+                json.dump(data, outfile)
+        # Print the results in the CLI.
+        else:
+            if test_result_count:
+                print(f"\n{test_result_count} abnormally slower tests:")
+                for func_name, timing in test_results:
+                    time = float(timing)
+                    baseline = float(self.baselines[func_name])
+                    print(f"{time:.4f}s {func_name} is significantly slower than the baseline {baseline:.4f}s")
+                # Make the test fail if there are tests significantly slower
+                assert test_result_count == 0, "Performance Tests failed due to significantly slower tests"
+
+            if not test_results:
+                print("\nNo tests signficantly slower than baseline. Success!")
+
+    def get_baselines(self):
+        """Load the performance_baselines.yml file for result comparison."""
+        baselines = {}
+        input_file = "nautobot/core/tests/performance_baselines.yml"
+
+        with open(input_file) as f:
+            data = yaml.load(f, Loader=SafeLoader)
+            for entry in data["tests"]:
+                baselines[entry["name"]] = entry["execution_time"]
+        return baselines
+
+    def suite_result(self, suite, result):
+        """Compile the performance test results"""
+        return_value = super(DiscoverSlowestTestsRunner, self).suite_result(suite, result)
+        self.baselines = self.get_baselines()
+
+        # You can set `ALWAYS_GENERATE_SLOW_REPORT` to True or add `--report` to `invoke` commands to generate report.
+        # e.g. `invoke unittest --report`
+        should_generate_report = getattr(settings, "ALWAYS_GENERATE_SLOW_REPORT", True) or self.should_generate_report
+        if not should_generate_report:
+            self.remove_timing_tmp_files()
```

Comment:
Where's this method defined?

Reply:
This method is defined in `DiscoverSlowestTestsRunner` class inherited by `NautobotTestRunner`.
See https://github.com/realpython/django-slow-tests/blob/2c5673ca5b6eb093a908e4e28143d889a1893113/django_slowtests/testrunner.py#L219